src/balance.js

```diff
@@ -1,17 +1,33 @@
 const { BN } = require('./setup');
 
-async function balanceCurrent (account) {
-  return new BN(await web3.eth.getBalance(account));
+class Tracker {
+  constructor (acc) {
+    this.account = acc;
+    this.current = async () => new BN(await web3.eth.getBalance(this.account));
+  };
+  async delta () {
+    const current = await this.current();
+    const delta = current.sub(this.prev);
+    this.prev = current;
+    return delta;
+  }
+  async get () {
+    this.prev = await this.current();
+    return this.prev;
+  }
+}
+
+async function balanceTracker (owner) {
+  const tracker = new Tracker(owner);
+  await tracker.get();
```

Comment:
I think it might make more sense to put this inside of the `Tracker` constructor, making the class itself safer to use.

Reply:
You mean the first call to `get()`? Or export the whole classe and get rid of `balanceTracker` altogether?

Reply:
Yes, the call to get. Exporting the class might make sense, it looks like the only difference would be having to use `new`, right? That may help people understand that the tracker is a proper object, but I'll leave it up to you.

Reply:
I think the main issue is that constructors cannot be `async`, so that would leave the call to `get()` hanging. Although it would be fine on almost all the cases, it's kind of fragile.The same issue happens when exporting the whole class, since it involves a async call to check the balance, the usage gets a bit weird, and inconsistent with `balance.current`.

One solution is to remove that first call completely and instruct users to always call `get()` before starting calling `delta()`, but I think it makes usage unnecessarily complicated.

Considering the trade offs, I would leave as is, but I'm not a heavy user of javascript so I might be missing some better solution.

Reply:
Ah, I didn't remember about not being able to have `async` constructors. Let's leave it as is then, thanks!